postgres/src/main/java/org/jdbi/v3/postgres/JavaTimeArgumentFactory.java

```diff
@@ -31,22 +31,20 @@
  * Note that no {@link java.time.Instant} override is needed.
  */
 public class JavaTimeArgumentFactory implements ArgumentFactory {
-
     @Override
     public Optional<Argument> build(Type type, Object value, ConfigRegistry config) {
+        final int sqlType;
         if (type == LocalDate.class) {
-            return Optional.of(new ObjectArgument(value, Types.DATE));
-        }
-        if (type == LocalTime.class) {
-            return Optional.of(new ObjectArgument(value, Types.TIME));
-        }
-        if (type == LocalDateTime.class) {
-            return Optional.of(new ObjectArgument(value, Types.TIMESTAMP));
+            sqlType = Types.DATE;
+        } else if (type == LocalTime.class) {
+            sqlType = Types.TIME;
+        } else if (type == LocalDateTime.class) {
+            sqlType = Types.TIMESTAMP;
+        } else if (type == OffsetDateTime.class) {
+            sqlType = Types.TIMESTAMP_WITH_TIMEZONE;
+        } else {
+            return Optional.empty();
         }
-        if (type == OffsetDateTime.class) {
-            return Optional.of(new ObjectArgument(value, Types.TIMESTAMP_WITH_TIMEZONE));
-        }
-        return Optional.empty();
+        return Optional.of(ObjectArgument.of(value, sqlType));
```

Comment:
This structure smells like I'd delegate it to a `Map<Class, Integer>` instead of a code tree if you're gonna refactor it anyway